active/0000-store-fetch.md

```diff
@@ -0,0 +1,282 @@
+sure we give developers the tools to build UIs that can indicate to the
+user that the information is being updated, perhaps by greying it out or
+displaying a loading spinner.
+
+# Detailed design
+
+## Proposal
+
+### New Adapter Methods.
+
+`shouldRefetchRecord` is a new method on the adapter that will be called by the store to make initial decision whether to refetch the record form the adapter or return the cached record from the store. This could method could be used to implement caching logic (e.g. only refetch this record after the time specified in its cache expires token) or for improved offline support (e.g. always refetch unless there is no internet connection then use cached record).
+
+This record would only be called if the record was already found in the store and is in the loaded state. 
+
+This method is only checked by `store.findById` and `store.findAll`. Methods with `fetch` in their name always refetch the record(s) from the adapter.
+
+```js
+{
+  /**
+   `shouldRefetchRecord` returns true if the adapter determines the record is
+   stale and should be refetch. It should return false if the record
+   is not stale or some other condition indicates that a fetch should
+   not happen at this time (e.g. loss of internet connection). 
+
+   This method is synchronous.
+
+   @method shouldRefetchRecord
+   @param {DS.Store} store
+   @param {DS.Model} record
+   @param {Object} adapterOptions
+   @return {Boolean}
+   */
+  shouldRefetchRecord: function(store, record, adapterOptions),
+}
+```
+
+The method `shouldBackgroundUpdate` would be used by the store to make the decision to re-fetch the record after it has already been returned to the user. This would allow realtime adapter to opt out of the background fetch if the adapter is already subscribing to changes on the record. 
+
+```js
+{
+  /**
+   `shouldBackgroundUpdate` returns true if the store should re fetch a
+   record in the store after returning it to the user to ensure the
+   record has the most up to date data.
+   
+   This method is synchronous.
+
+   @method shouldBackgroundUpdate
+   @param {DS.Store} store
+   @param {DS.Model} record
+   @param {Object} adapterOptions
+   @return {Boolean}
+  */
+  shouldBackgroundUpdate: function(store, record, adapterOptions),
+}
+```
+
+
+In the next major version of Ember Data the recommend way of finding a record 
+will be:
+
+```js
+this.store.findById('person', 1);
+```
+
+This will return a promise that:
+
+1. Waits to resolve until the data is fetched from the server, on the
+   initial request.
+2. Resolves immediately with the locally cached request for subsequent
+   requests, but triggers a request to the server for the updated
+   version and updates the record in-place if there are changes.
+
+
+In terms of the above methods `shouldRefetchRecord` will always return `false` and `shouldBackgroundUpdate` will always return `true` in the default `RESTAdapter`.
```

Comment:
Maybe it's because I'm pretty used to ember-data current behaviour, but I think that return false from `shouldBackgroundUpdate` would be a better default.

It is retrocompatible with what ED does now and I feel that refetch information that I already have is something you should opt-in to instead of opt-out.